erpnext/assets/doctype/asset_repair/asset_repair.py

```diff
@@ -5,14 +5,172 @@
 from frappe.model.document import Document
+from erpnext.accounts.general_ledger import make_gl_entries
 
 class AssetRepair(Document):
 	def validate(self):
 		if self.repair_status == "Completed" and not self.completion_date:
-			frappe.throw(_("Please select Completion Date for Completed Repair"))
+			self.completion_date = nowdate()
 
+		self.update_status()
+		self.set_total_value()		# change later
+		self.calculate_total_repair_cost()
+		
+	def update_status(self):
+		if self.repair_status == 'Pending':
+			frappe.db.set_value('Asset', self.asset, 'status', 'Out of Order')
+		else:
+			asset = frappe.get_doc('Asset', self.asset)
+			asset.set_status()
+
+	def set_total_value(self):
+		for item in self.stock_items:
+			item.total_value = flt(item.valuation_rate) * flt(item.consumed_quantity)
+
+	def calculate_total_repair_cost(self):
+		self.total_repair_cost = self.repair_cost
+		for item in self.stock_items:
+			self.total_repair_cost += item.total_value
+
+	def on_submit(self):
+		self.check_repair_status()
+		self.check_for_cost_center()
+
+		if self.stock_consumption or self.capitalize_repair_cost:
+			self.increase_asset_value()
+		if self.stock_consumption:
+			self.check_for_stock_items_and_warehouse()
+			self.decrease_stock_quantity()
+		if self.capitalize_repair_cost:
+			self.check_for_purchase_invoice()
+			self.make_gl_entries()
+			if frappe.db.get_value('Asset', self.asset, 'calculate_depreciation'):
+				self.modify_depreciation_schedule()
+
+	def check_repair_status(self):
+		if self.repair_status == "Pending":
+			frappe.throw(_("Please update Repair Status."))
+
+	def check_for_stock_items_and_warehouse(self):
+		if not self.stock_items:
+			frappe.throw(_("Please enter Stock Items consumed during Asset Repair."))
+		if not self.warehouse:
+			frappe.throw(_("Please enter Warehouse from which Stock Items consumed during Asset Repair were taken."))
+
+	def check_for_cost_center(self):
+		if not self.cost_center:
+			frappe.throw(_("Please enter Cost Center."))
+
+	def increase_asset_value(self):
+		asset_value = frappe.db.get_value('Asset', self.asset, 'asset_value')
+		for item in self.stock_items:
+			asset_value += item.total_value
+
+		if self.capitalize_repair_cost:
+			asset_value += self.repair_cost
+		frappe.db.set_value('Asset', self.asset, 'asset_value', asset_value)
+
+	def decrease_stock_quantity(self):
+		stock_entry = frappe.get_doc({
+			"doctype": "Stock Entry",
+			"stock_entry_type": "Material Issue",
+			"company": frappe.get_value('Asset', self.asset, "company")
+		})
+
+		for stock_item in self.stock_items:
+			stock_entry.append('items', {
+				"s_warehouse": self.warehouse,
+				"item_code": stock_item.item,
+				"qty": stock_item.consumed_quantity
+			})
+
+		stock_entry.insert()
+		stock_entry.submit()
+
+	def check_for_purchase_invoice(self):
+		if not self.purchase_invoice:
+			frappe.throw(_("Please link Purchase Invoice."))
```

Comment:
Use `mandatory_depends_on` here too, the condition would be `repair_status == 'Completed' && repair_cost > 0`

Reply:
Done.